File: CSES/DynamicProgramming/MinimizingCoins.cpp
#include <iostream>
#include <vector>
#include <limits>

using namespace std;

int main() {
    int n, x;
    cin >> n >> x;

    vector<int> coins(n);
    for (int i = 0; i < n; i++) {
        cin >> coins[i];
    }

    vector<int> dp(x + 1, 1000001);
    dp[0] = 0;

    for (int i = 1; i <= x; i++) {
        for (int coin : coins) {
            if (i >= coin) {
                dp[i] = min(dp[i], dp[i - coin] + 1);
            }
        }
    }

    if (dp[x] == 1000001) {
        cout << -1 << endl;
    } else {
        cout << dp[x] << endl;
    }

    return 0;
}

File: CSES/DynamicProgramming/MinimizingCoins.py
from math import isinf

n,x = tuple(map(int,input().split()))
coins = list(map(int,input().split()))

dp = [float('INF')]*(x+1)
dp[0] = 0

for i in range(1,x+1):
    for coin in coins:
        if i>=coin:
            dp[i] = min(dp[i],dp[i-coin]+1)

if isinf(dp[-1]):
    print(-1)
else:
    print(dp[-1])